tools/gulp/util/firebase.ts

```diff
@@ -6,18 +6,26 @@ const cloudStorage = require('@google-cloud/storage');
 const screenshotFirebaseConfig = require('../../screenshot-test/functions/config.json');
 
 /** Opens a connection to the Firebase dashboard app. */
-export function openFirebaseDashboardApp() {
+export function openFirebaseDashboardApp(asGuest = false) {
+  const databaseURL = 'https://material2-board.firebaseio.com';
+
+  // In some situations the service account credentials are not available and authorizing as
+  // a guest works fine. For example in Pull Requests the payload task just wants to read data.
+  if (asGuest) {
+    return firebase.initializeApp({ databaseURL });
+  }
```

Comment:
I'd just make this a different function: `openFirebaseDashboardAppAsGuest`

(same thing about flag arguments https://martinfowler.com/bliki/FlagArgument.html)